test/expr/unary/keypath/keypath.swift

```diff
@@ -741,3 +741,17 @@ func testSyntaxErrors() { // expected-note{{}}
   _ = \A.a?;
   _ = \A.a!;
 } // expected-error@+1{{}}
+
+class SR_10146 {
+  let abc: Int = 1
+  
+  func doNotCrash() {
+    let _: KeyPath<AnyObject, Int> = \.1 // expected-error {{type 'AnyObject' has no member '1'}}
+    let _: KeyPath<AnyObject, Int> = \.abc // expected-error {{type 'AnyObject' has no member 'abc'}}
+  }
+
+  func doNotCrash_1(_ obj: AnyObject, _ kp: KeyPath<AnyObject, Int>) {
+    let _ = obj[keyPath: \.abc] // expected-error {{type of expression is ambiguous without more context}}
+    let _ = obj[keyPath: kp]
```

Comment:
This should shouldn't be accepted right? The root type is `AnyObject` in this case.

Reply:
This new tests should probably be migrated to keypath-objc.swift otherwise it's just going to trigger missing member always...

Reply:
Hmm I think so, but we aren't calling `simplifyKeyPathConstraint()` in this case (`obj[keyPath: kp]`).